optaplanner-core/src/main/java/org/optaplanner/core/impl/score/director/stream/ConstraintStreamScoreDirector.java

```diff
@@ -82,7 +82,7 @@ public boolean isConstraintMatchEnabled() {
     }
 
     @Override
-    public Map<String, ConstraintMatchTotal> getConstraintMatchTotalMap() {
+    public <Score_ extends Score<Score_>> Map<String, ConstraintMatchTotal<Score_>> getConstraintMatchTotalMap() {
```

Comment:
This is probably the only thing that makes me a little bit nervous: the type definition is duplicated over multiple methods, effectively being a different type each time.

Reply:
True. I can go deeper and add `<Score_>` on more and more classes, but this is the MVP that we agreed I'd do.

Reply:
Indeed, a Score_ on InnerScoreDirector can fix that, but the side-effects of that can be severe.